framework/db/src/main/java/com/cloud/utils/db/TransactionLegacy.java

```diff
@@ -1143,6 +1134,38 @@ public static void initDataSource(Properties dbProps) {
         }
     }
 
+    private static String getConnectionUri(Properties dbProps, String loadBalanceStrategy, String driver, boolean useSSL, String databaseName) {
+        String connectionUri;
+        String uri = dbProps.getProperty(String.format("db.%s.uri", databaseName));
+
+        if (StringUtils.isEmpty(uri)) {
+            String host = dbProps.getProperty(String.format("db.%s.host", databaseName));
+            int port = Integer.parseInt(dbProps.getProperty(String.format("db.%s.port", databaseName)));
+            String dbName = dbProps.getProperty(String.format("db.%s.name", databaseName));
+            boolean autoReconnect = Boolean.parseBoolean(dbProps.getProperty(String.format("db.%s.autoReconnect", databaseName)));
+            String url = dbProps.getProperty(String.format("db.%s.url.params", databaseName));
+
+            String replicas = null;
+            String dbHaParams = null;
+            if (s_dbHAEnabled) {
+                dbHaParams = getDBHAParams(databaseName, dbProps);
+                replicas = dbProps.getProperty(String.format("db.%s.replicas", databaseName));
+                s_logger.info(String.format("The replicas configured for %s data base are %s.", databaseName, replicas));
+            }
+
+            connectionUri = driver + "://" + host + (s_dbHAEnabled ? "," + replicas : "") + ":" + port + "/" + dbName +
+                    "?autoReconnect=" + autoReconnect + (url != null ? "&" + url : "") + (useSSL ? "&useSSL=true" : "") +
+                    (s_dbHAEnabled ? "&" + dbHaParams : "") + (s_dbHAEnabled ? "&loadBalanceStrategy=" + loadBalanceStrategy : "");
+        } else {
+            s_logger.warn(String.format("db.%s.uri was set, only using the following properties of db.properties: [maxActive, maxIdle, maxWait, username, password, driver, "
```

Comment:
This information is incorrect; we have a bunch of other properties that still will be used. It would be clearer if we would inform the properties that will be ignored.